onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -0,0 +1,709 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/propagate_cast_ops.h"
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/utils.h"
+#include <deque>
+
+using namespace ONNX_NAMESPACE;
+using namespace onnxruntime::common;
+namespace onnxruntime {
+// The collection fp16_allow_ops, specifies for a given propagate_cast_ops level, a vector of node op_types that
+// the code is allowed to propage Cast operations cross. The user may specify a custom list of optypes using level 0.
+// The opcodes are split into multiple levels. Cast propagation is done based on the level. Level 2 op code
+// list includes Level 1 list also.
+static std::vector<std::unordered_set<std::string>> fp16_allow_ops = {
+    /* Level 0 */ {},
+    /* Level 1 */ {"Transpose", "Reshape", "Split", "Relu", "BiasFastGelu", "BiasGelu", "LayerNorm", "Gelu", "FastGelu", "Tanh"},
```

Comment:
>LayerNorm [](start = 89, length = 9)

LayerNorm should be in Level 2, since our current kerenl changes the underlaying computation percisioon

Reply:
Also, the need to use the full name for LayerNorm here. 
LayerNormalization, 

---
In reply to: [611998492](https://github.com/microsoft/onnxruntime/pull/7191#discussion_r611998492) [](ancestors = 611998492)